src/account/PluginManager2.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {IPlugin} from "../interfaces/IPlugin.sol";
+import {FunctionReference} from "../interfaces/IPluginManager.sol";
+import {FunctionReferenceLib} from "../helpers/FunctionReferenceLib.sol";
+import {AccountStorage, getAccountStorage, toSetValue} from "./AccountStorage.sol";
+
+// Temporary additional functions for a user-controlled install flow for validation functions.
+abstract contract PluginManager2 {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    error DefaultValidationAlreadySet(address plugin, uint8 functionId);
+    error ValidationAlreadySet(bytes4 selector, address plugin, uint8 functionId);
+    error ValidationNotSet(bytes4 selector, address plugin, uint8 functionId);
+
+    function _installValidation(
+        address plugin,
+        uint8 functionId,
+        bool isDefault,
```

Comment:
Is this necessary? Is the goal to separate validation installation from other plugin installations?

Would it be easier to only separate default validation installation from all other plugin installations (include non default validation installations).